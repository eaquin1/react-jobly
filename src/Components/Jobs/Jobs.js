import React, { useEffect, useState } from "react";
import Search from "../Search/Search";
import JoblyApi from "../../Helpers/JoblyApi";
import JobCard from "../Cards/JobCard";
import { Link } from "react-router-dom";

function Jobs() {
    const [jobs, setJobs] = useState([]);

    useEffect(() => {
        async function getJs() {
            let j = await JoblyApi.getJobs();
            setJobs(j);
        }
        getJs();
    }, []);

    async function handleSearch(search) {
        let searchResults = await JoblyApi.getJobs(search);
        setJobs(searchResults);
    }

    async function apply(idx) {
        let jobId = jobs[idx].id;
        let message = await JoblyApi.applyToJob(jobId);
        setJobs((j) => {
            j.map((job) =>
                job.id === jobId ? { ...job, state: message } : job
            );
        });
    }
    // if (!jobs) {
    //     return <div>Loading...</div>;
    // }

  let none =  
       (
        <div>
            Job was not found. Please return back to
            <Link to="/jobs" onClick={() => handleSearch()}>
                {" "}
                Jobs{" "}
            </Link>
        </div>)
   
   let found = jobs.map((job, idx) => (
    <JobCard
        item={job}
        key={idx}
        idx={idx}
        handleApply={() => apply(idx)}
    />
))

    return  (
        <div>
            <Search endpoint="jobs" searchFor={handleSearch} />
            <div>{jobs.length ? found : none} </div>
        </div>
    );
}

export default Jobs;
