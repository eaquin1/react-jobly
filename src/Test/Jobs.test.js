import React from "react";
import { render } from "@testing-library/react";
import Jobs from "../Components/Jobs/Jobs";
import {MemoryRouter} from "react-router"

it("renders without crashing", function() {
    
  render(<MemoryRouter>
  <Jobs />
  </MemoryRouter>);
  
});

it("matches snapshot with no jobs", function() {
  const { asFragment } = render(
    <MemoryRouter>
  <Jobs />
  </MemoryRouter>);
  expect(asFragment()).toMatchSnapshot();
});