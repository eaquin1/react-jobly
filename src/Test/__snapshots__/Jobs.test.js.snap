// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with no jobs 1`] = `
<DocumentFragment>
  <div>
    <div
      style="margin-left: 40px;"
    >
      <form
        class="form-inline"
      >
        <input
          class="form-control"
          name="search"
          placeholder="Enter a search term"
          type="text"
          value=""
        />
        <button
          class="btn btn-secondary"
        >
          Submit
        </button>
      </form>
    </div>
    <div>
      <div>
        Job was not found. Please return back to
        <a
          href="/jobs"
        >
           Jobs 
        </a>
      </div>
       
    </div>
  </div>
</DocumentFragment>
`;
